### PR TITLE
Allow disable reporting to be controlled from the config file

### LINKED ISSUE #3858: Add support for reportNeedlessDisables to the configuration object
> What is the problem you're trying to solve?

I always want to use the `--report-needless-disables` flag, but it's a chore having to use it for every `$ stylelint` invocation.

> What solution would you like to see?

I would like the ability to specify the CLI flag as a setting in the Stylelint config. That way I can put it in my shareable config and I don't have to specify it each time. The setting could be named `reportNeedlessDisables`.

### PR DESCRIPTION
**Pull request by nex3**
This will also make it easier to control disables on a file-by-file basis
once #3128 is implemented. It will also make it possible to add 
finer-grained configuration to these rules in the future.

Closes #3858

### CODE REVIEW

File: lib/needlessDisables.js [hunk 1]
```diff
@@ -72,7 +78,7 @@ module.exports = function (results) {
 
 				// Only emit a warning if this range's comment isn't useful for this rule.
 				// For the special rule "all", only emit a warning if it's not useful for
-				// *any* // rules, becuase it covers all of them.
+				// *any* rules, becuase it covers all of them.
```
Comment (jathak) [inline lib/needlessDisables.js:81]:
```suggestion
				// *any* rules, because it covers all of them.
```

**Review by jathak — APPROVED** [review]

**Review by vankop — APPROVED** [review]

File: docs/user-guide/configure.md [hunk 2]
```diff
@@ -168,6 +168,84 @@ You can set the default severity level for all rules that do not have a severity
 }
 ```
 
+## `ignoreDisables`
+
+Ignore `styleline-disable` (e.g. `/* stylelint-disable block-no-empty */`) comments.
```
Comment (jeddy3) [inline docs/user-guide/configure.md:173]:
```suggestion
Ignore `stylelint-disable` (e.g. `/* stylelint-disable block-no-empty */`) comments.
```

File: docs/user-guide/configure.md [hunk 3]
```diff
@@ -168,6 +168,84 @@ You can set the default severity level for all rules that do not have a severity
...
+
+```json
+{
+  "ignoreDisables": true
+}
+```
+
+## `reportNeedlessDisables`
+
+Emit errors for `stylelint-disable` comments that don't include actually match
```
Comment (jeddy3) [inline docs/user-guide/configure.md:185]:
```suggestion
Emit errors for `stylelint-disable` comments that don't match
```

File: docs/user-guide/configure.md [hunk 4]
```diff
@@ -168,6 +168,84 @@ You can set the default severity level for all rules that do not have a severity
...
+any lints that need to be disabled.
```
Comment (jeddy3) [inline docs/user-guide/configure.md:186]: Let's remove the hard linebreak to be consistent with the rest of the file

File: docs/user-guide/configure.md [hunk 5]
```diff
@@ -168,6 +168,84 @@ You can set the default severity level for all rules that do not have a severity
...
+```json
+{
+  "reportNeedlessDisables": true
+}
+```
+
+## `reportInvalidScopeDisables`
+
+Emit errors for `stylelint-disable` comments that don't match rules that are
+specified in the configuration object.
```
Comment (jeddy3) [inline docs/user-guide/configure.md:199]:
```suggestion
Emit errors for `stylelint-disable` comments that don't match rules that are specified in the configuration object.
```

**Review by jeddy3 — CHANGES_REQUESTED** [review]
@nex3 Thank you. It's amazing to see this go in. As you said, it's the first step towards adding the `overrides` configuration property (which opens up so many possibilities). It's also the first step towards being able to [configure everything](https://github.com/stylelint/stylelint/issues/4843#issuecomment-650137005) from within the configuration object, rather than a hodgepodge of properties, parameters and flags.

I've requested some minor documentation changes.

At the start of last year we [restructured](https://github.com/stylelint/stylelint/pull/4548) and [rewrote](https://github.com/stylelint/stylelint/pull/4561) the majority of the docs. This included removing a lot of duplication where API parameters, CLI flags and configuration properties had fallen out of sync. 

I'm aware this pull request reintroduces duplication, but I suggest we roll with it for now. I'll also merge in https://github.com/stylelint/stylelint/pull/5005, which duplicates the CLI flags. Once everything is configurable within the configuration object, I think we'll be better positioned to revisit the documentation and find a structure that both avoids duplication and is intuitive for users.

**Review by jeddy3 — APPROVED** [review]
Thanks for making the requested changes.

LGTM!

**Comment by jeddy3** [comment]
> - Added: `ignoreDisables`, `reportNeedlessDisables`, `reportInvalidScopeDisables` and `reportDescriptionlessDisables` configuration object properties ([#5126](https://github.com/stylelint/stylelint/pull/5126)).

**Pull request merged**